base/ltclass.dtx

```diff
@@ -1520,6 +1521,7 @@
    \ifx\@classoptionslist\relax\else
     \@for\CurrentOption:=\@classoptionslist\do{%
       \ifx\CurrentOption\@empty\else
+        \edef\CurrentOption{\detokenize\expandafter{\CurrentOption}}%
```

Comment:
Using `\detokenize` here in re-defining `\CurrentOption` means that this 'shows up' in `\@use@ption`: if we imagine for example an option called `foo{=}bar`, the detokenization would now strip the braces of meaning, and we'd get 'unused option `foo`' not 'unused option `foo{=}bar`'. I can see that is somewhat contrived, but want to be sure that the effect is deliberate/expected.

Reply:
@josephwright  yes but this is the classic option handler and as existing behaviour is the low level csname error, there don't seem to be any compatibility issues, in practice only simple names are going to be declared, the only issue is to make sure foo=\whatever survives eough to get on the global class options list (without being detokenized)

Reply:
adjusted